frontend/src/components/utils.js

```diff
@@ -8,124 +8,154 @@ export const flatData = (data) => {
   return ratingSets.every((curr) => curr.toString() === ratingSets[0].toString());
 };
 
-export const pivotHeatMapData = (payload) => {
-  // this transformation takes into account that not all criteria have the same rating set
+/**
+ * Return a map of objects with sectionId and formatted sectionName
+ * @param {array} sections
+ * @returns array of objects
+ */
+export const sectionIdNameMap = (sections) => sections.map((s) => ({
+  sectionId: s.sis_section_id,
+  sectionName: s.name.split(' ').pop(),
+}));
+
+/**
+ * filter test to filter for unique student_id and criterion_id combinations
+ * @param {object} curr - the current object
+ * @param {int} index - the current index in the original array
+ * @param {array} arr - the original array
+ * @returns boolean
+ */
+const uniqueStudentCriteria = (curr, index, arr) => {
+  const firstIndex = arr.findIndex((item) => (
+    item.student_id === curr.student_id
+      && item.criterion_id === curr.criterion_id
+  ));
+  return firstIndex === index;
```

Comment:
Yeah, there are a couple of places that could be optimized that I am storing in my head. Maybe I should add some TODO comments.
As far as this function is concerned, it was handling a class roster of 111 in miliseconds, but the whole filter function where this is used a callback is O(n^2), so it could definitely be a bottleneck. I'm not so up on algorithm efficiencies and would love to dojo some improvements.